daemon/cmd/ipam.go

```diff
@@ -266,17 +266,22 @@ func (d *Daemon) allocateDatapathIPs(family types.NodeAddressingFamily, fromK8s,
 	// by the caller.
 	// This will also cause disruption of networking until all endpoints
 	// have been regenerated.
+	primaryFam := ipam.IPv4
```

Comment:
This looks off: If we are restoring a router IP from before a cilium-agent restart, `allocateDatapathIPs` is called with `fromFS` being a valid IPv6. Thus the family of the allocated IP would be IPv6. We ought to check the family type of `fromK8s/fromFS` here